packages/server/api/src/app/mcp/mcp-server/mcp-server.ts

```diff
@@ -71,32 +79,40 @@ async function addPieceToServer(
         platformId,
     })
 
-    const filteredAction = Object.keys(pieceMetadata.actions).filter(action => toolPieceMetadata.actionNames.includes(action))
-    for (const action of filteredAction) {
+    const toolActionsNames = Object.keys(pieceMetadata.actions).filter(action => toolPieceMetadata.actionNames.includes(action))
+    for (const action of toolActionsNames) {
         const actionMetadata = pieceMetadata.actions[action]
         const actionName = mcpToolNaming.fixTool(actionMetadata.displayName, mcpTool.id, McpToolType.PIECE)
 
-        const actionSchema = Object.fromEntries(
-            Object.entries(actionMetadata.props)
-                .filter(([_key, prop]) => prop.type !== PropertyType.MARKDOWN)
-                .map(([key, prop]) => [key, piecePropertyToZod(prop)]),
-        )
+        const toolSchema = {
+            instructions: z.string().describe('The instructions to follow when executing the tool'),
```

Comment:
I think the description is useless there, I think It should be something very specific like what to include and how detailed is it.